api/receiver/appengine.go

```diff
@@ -19,10 +19,18 @@ import (
 	"google.golang.org/appengine/user"
 )
 
+// DatastoreKey is a context-free constructable form of datastore.Key that
+// contains identifying data for a datastore object that was assigned an
+// integral ID with no parent key.
```

Comment:
Oh right, `datastore.Key` has a context... Sad. I might choose returning only the `ID`, but this is good, too.

Reply:
I initially returned the ID, but we had tests that verify that the kind we get back is `TestRun`. Believing that the tests are meaningful, I changed the code to preserve both pieces of data from the key that might be relevant.